src/routes/consent.ts

```diff
@@ -0,0 +1,132 @@
+import { NextFunction, Request, Response } from 'express'
+import hydra from './../services/hydra.js';
+import url from 'url';
+
+export const getConsent = (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+  // Parses the URL query
+  var query = url.parse(req.url, true).query;
+  // The challenge is used to fetch information about the consent request from ORY Hydra.
+  var challenge = query.consent_challenge;
+
+  hydra.getConsentRequest(challenge)
+  // This will be called if the HTTP request was successful
+    .then(function (response:any) {
+      // If a user has granted this application the requested scope, hydra will tell us to not show the UI.
+      if (response.skip) {
+        // You can apply logic here, for example grant another scope, or do whatever...
+
+        // Now it's time to grant the consent request. You could also deny the request if something went terribly wrong
+        return hydra.acceptConsentRequest(challenge, {
+          // We can grant all scopes that have been requested - hydra already checked for us that no additional scopes
+          // are requested accidentally.
+          grant_scope: response.requested_scope,
+
+          // ORY Hydra checks if requested audiences are allowed by the client, so we can simply echo this.
+          grant_access_token_audience: response.requested_access_token_audience,
+
+          // The session allows us to set session data for id and access tokens
+          session: {
+            // This data will be available when introspecting the token. Try to avoid sensitive information here,
+            // unless you limit who can introspect tokens.
+            // access_token: { foo: 'bar' },
+
+            // This data will be available in the ID token.
+            // id_token: { baz: 'bar' },
+          }
+        }).then(function (response:any) {
+          // All we need to do now is to redirect the user back to hydra!
+          res.redirect(response.redirect_to);
+        });
+      }
+
+      // If consent can't be skipped we MUST show the consent UI.
+      res.render('consent', {
+        // csrfToken: req.csrfToken() ? req.csrfToken() : 'none',
+        // csrfToken: TODO CSRF? 'no CSRF!',
```

Comment:
This endpoint should probably be csrf-protected

Reply:
implemented in https://github.com/ory/kratos-selfservice-ui-node/pull/50/commits/10c2d6075bca352d7bf258be7e6a76003d83f764